memory/table.go

```diff
@@ -1907,8 +1907,7 @@ func (t *Table) createIndex(data *TableData, name string, columns []sql.IndexCol
 		}
 	}
 	if data.indexes[name] != nil {
-		// TODO: extract a standard error type for this
-		return nil, fmt.Errorf("Error: index already exists")
+		return nil, sql.ErrDuplicateKey.New(name)
```

Comment:
Looks like the error message changed; was "index already exists" not accurate or are we just trying to match MySQL's message closer? If the latter... the error message I got from MySQL when I tried to create a new index using an existing index name was: 
```
`foo` already exists as an index for this table
```

Reply:
this is the output I got from mysql
```sql
mysql> create table t (i int, index (i));
Query OK, 0 rows affected (0.0254 sec)
mysql> create index i on t(i);
ERROR: 1061: Duplicate key name 'i'
```